shared/ui/Stream/PullRequests/Bitbucket/PullRequest.tsx

```diff
@@ -97,389 +94,150 @@ export const PullRequestRoot = styled.div`
 	const [editingTitle, setEditingTitle] = useState(false);
 	const [savingTitle, setSavingTitle] = useState(false);
 	const [title, setTitle] = useState("");
+	const [currentRepoChanged, setCurrentRepoChanged] = useState(false);
 	const [finishReviewOpen, setFinishReviewOpen] = useState(false);
-	const [dynamicKey, setDynamicKey] = useState("");
+	const [oneLayerModal, setOneLayerModal] = useState(false);
+	const [autoCheckedMergeability, setAutoCheckedMergeability] =
+		useState<autoCheckedMergeabilityStatus>("UNCHECKED");
 	const [prCommitsRange, setPrCommitsRange] = useState<string[]>([]);
-	const [isMerging, setIsMerging] = useState(false);
-	const [isDeclining, setIsDeclining] = useState(false);
-	const [isMergingStrategy, setIsMergingStrategy] = useState("");
 
-	const breakpoints = {
-		auto: "630px",
-		"side-by-side": "10px",
-		vertical: "100000px",
+	const switchActiveTab = tab => {
+		// remember the scroll position of the tab we just left
+		const container = document.getElementById("pr-scroll-container");
+		if (container) setScrollPosition({ ...scrollPosition, [activeTab]: container.scrollTop });
+		setActiveTab(tab);
 	};
-	const addViewPreferencesToTheme = theme => ({
-		...theme,
-		breakpoint: breakpoints[derivedState.viewPreference],
-	});
-
-	const closeFileComments = () => {
-		// note we're passing no value for the 3rd argument, which clears
-		// the commentId
-		// if (pr) dispatch(setCurrentPullRequest(pr.providerId, pr.idComputed));
-		if (oneLayerModal && pr) {
-			dispatch(setCurrentPullRequest(pr.providerId, pr.idComputed, "", "", "sidebar-diffs"));
-		}
 
-		if (!oneLayerModal && pr) {
-			dispatch(setCurrentPullRequest(pr.providerId, pr.idComputed, "", "", "details"));
-		}
+	const exit = async () => {
+		await dispatch(clearCurrentPullRequest());
 	};
 
-	const _assignState = _pr => {
-		if (!_pr) return;
-		// if (!_pr.project) {
-		// 	console.warn("possible bad request");
-		// }
+	// const [reviewText, setReviewText] = useState("");
+	const [submittingReview, setSubmittingReview] = useState(false);
+	const [reviewType, setReviewType] = useState<
+		"MERGE" | "APPROVE" | "UNAPPROVE" | "REQUEST_CHANGES"
+	>("APPROVE");
 
-		//	if (_pr && _pr.project) setTitle(_pr.project.mergeRequest.title);
+	const PRError = styled.div`
+		padding: 0px 15px 20px 15px;
+		display: flex;
+		align-items: center;
+		> .icon {
+			flex-grow: 0;
+			flex-shrink: 0;
+			display: inline-block;
+			margin-right: 15px;
+			transform: scale(1.5);
+			color: #ff982d;
+		}
+		> div {
+			color: #ff982d;
+			flex-grow: 10;
+			display: flex;
+			align-items: center;
+			button {
+				margin-left: auto;
+			}
+		}
+		strong {
+			font-weight: normal;
+			color: var(--text-color-highlight);
+		}
+		a {
+			text-decoration: none;
+			color: var(--text-color-highlight);
+			&:hover {
+				color: var(--text-color-info) !important;
+			}
+		}
+	`;
+
+	const _assignState = (pr, src?: string) => {
```

Comment:
Can be typed go go along with `useState`

```suggestion
  const _assignState = (pr: FetchThirdPartyPullRequestResponse, src?: string) => {
```